prism/src/main/java/org/apache/falcon/resource/extensions/ExtensionManager.java

```diff
@@ -294,36 +300,55 @@ public APIResult delete(@PathParam("job-name") String jobName,
 
     @POST
     @Path("submit/{extension-name}")
-    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.MULTIPART_FORM_DATA})
+    @Consumes({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.MULTIPART_FORM_DATA,
+            MediaType.APPLICATION_OCTET_STREAM})
     @Produces({MediaType.TEXT_XML, MediaType.TEXT_PLAIN, MediaType.APPLICATION_JSON})
     public APIResult submit(
             @PathParam("extension-name") String extensionName,
             @Context HttpServletRequest request,
             @DefaultValue("") @QueryParam("doAs") String doAsUser,
             @QueryParam("jobName") String jobName,
-            @FormDataParam("entities") List<Entity> entities,
+            @FormDataParam("processes") List<FormDataBodyPart> processForms,
+            @FormDataParam("feeds") List<FormDataBodyPart> feedForms,
             @FormDataParam("config") InputStream config) {
         checkIfExtensionServiceIsEnabled();
+        List<Process> processes = getProcesses(processForms);
+        List<Feed> feeds = getFeeds(feedForms);
+        Pair<List<Feed>, List<Process>> entityPairs;
```

Comment:
Map<EntityType, List<Entity>> is more extendable. Can be passed along directly as arg to submitEntities.